### PR TITLE
[`dbft`] Prevent Store Creation

### PR DESCRIPTION
**Pull request by cschuchardt88**
# Description
This PR prevents the directory and an empty store from being created when `IgnoreRecoveryLogs` is set to `true`.

## Type of change

- [ ] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Unit tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by vncoelho — COMMENTED** [review]
What is the advantage? As far as I see there is no fix.

**Review by vncoelho — COMMENTED** [review]
Ok, looks good to me. I will soon test.

File: src/Plugins/DBFTPlugin/Consensus/ConsensusContext.cs [hunk 1]
```diff
@@ -168,7 +170,7 @@ public Block EnsureHeader()
 
         public bool Load()
         {
-            byte[] data = store.TryGet(ConsensusStateKey);
+            byte[] data = store?.TryGet(ConsensusStateKey);
             if (data is null || data.Length == 0) return false;
```
Comment (shargon) [inline src/Plugins/DBFTPlugin/Consensus/ConsensusContext.cs:174]: if this return false it could affect `ConsensusService`. (I think that is not possible)

**Review by shargon — APPROVED** [review]

File: src/Plugins/DBFTPlugin/Consensus/ConsensusContext.cs (hunk 1 above)
Comment (Jim8y) [reply src/Plugins/DBFTPlugin/Consensus/ConsensusContext.cs:174]: Can we do something like change all Tryxxxx thing,,, make it return false or true

**Review by Jim8y — APPROVED** [review]

**Comment by Jim8y** [comment]
@vncoelho this pr will be blocked and wait for your review, please remove the blocked mark if you finished.

**Comment by Jim8y** [comment]
@vncoelho are you still reviewing it?

File: src/Plugins/DBFTPlugin/Consensus/ConsensusContext.cs (hunk 1 above)
Comment (cschuchardt88) [reply src/Plugins/DBFTPlugin/Consensus/ConsensusContext.cs:174]: This store is only for recovery.

https://github.com/neo-project/neo/blob/7e639a1318251d5da5eb237696d205e20719dea4/src/Plugins/DBFTPlugin/Consensus/ConsensusService.cs#L148-L162

**Pull request merged**